src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java

```diff
@@ -0,0 +1,230 @@
+        super.unload();
+        /* Disables custom spider. */
+        OpenApiSpider.disable();
+    }
+
+    /* Menu option to import a local OpenApi file. */
+    private ZapMenuItem getMenuImportLocalOpenApi() {
+        if (menuImportLocalOpenApi == null) {
+            menuImportLocalOpenApi = new ZapMenuItem("openapi.topmenu.tools.importOpenApi");
+            menuImportLocalOpenApi.setToolTipText(Constant.messages.getString("openapi.topmenu.tools.importOpenApi.tooltip"));
+
+            menuImportLocalOpenApi.addActionListener(new java.awt.event.ActionListener() {
+
+                @Override
+                public void actionPerformed(java.awt.event.ActionEvent e) {
+                    // Prompt for a OpenApi file.
+                    final JFileChooser chooser = new JFileChooser(Model.getSingleton().getOptionsParam().getUserDirectory());
+                    int rc = chooser.showOpenDialog(View.getSingleton().getMainFrame());
+                    if (rc == JFileChooser.APPROVE_OPTION) {
+                        importOpenApiDefinition(chooser.getSelectedFile());
+                    }
+
+                }
+            });
+        }
+        return menuImportLocalOpenApi;
+    }
+
+    /* Menu option to import a OpenApi file from a given URL. */
+    private ZapMenuItem getMenuImportUrlOpenApi() {
+        if (menuImportUrlOpenApi == null) {
+            menuImportUrlOpenApi = new ZapMenuItem("openapi.topmenu.tools.importRemoteOpenApi");
+            menuImportUrlOpenApi
+                    .setToolTipText(Constant.messages.getString("openapi.topmenu.tools.importRemoteOpenApi.tooltip"));
+
+            final ExtensionOpenApi shadowCopy = this;
+            menuImportUrlOpenApi.addActionListener(new java.awt.event.ActionListener() {
+
+                @Override
+                public void actionPerformed(java.awt.event.ActionEvent e) {
+                    SwingUtilities.invokeLater(new Runnable() {
+
+                        @Override
+                        public void run() {
+                            new ImportFromUrlDialog(View.getSingleton().getMainFrame(), shadowCopy);
+                        }
+                    });
+                }
+            });
+        }
+        return menuImportUrlOpenApi;
+    }
+
+    public void importOpenApiDefinition(final URI uri) {
+        Thread t = new Thread(THREAD_PREFIX + threadId++) {
+
+            @Override
+            public void run() {
+                Requestor requestor = new Requestor(HttpSender.MANUAL_REQUEST_INITIATOR);
+                requestor.addListener(new HistoryPersister());
+                try {
+                    importOpenApiDefinition(requestor.getResponseBody(uri));
+                } catch (SwaggerException e) {
+                    if (View.isInitialised()) {
+                        View.getSingleton().showWarningDialog(Constant.messages.getString("openapi.parse.error"));
+                    }
+                    LOG.error(e.getMessage(), e);
```

Comment:
Warn level instead (same for the other logs here)? We normally use error for possible bug/issues in the code (here is most likely bad input).